In [1]:
pip install statsmodels

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


In [2]:
pip install scipy

Defaulting to user installation because normal site-packages is not writeableNote: you may need to restart the kernel to use updated packages.



In [3]:
import numpy as np

## TASKS

## Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05
### Null Hypothesis
Ho = μ ≤ 11 minutes
Alternative Hypothesis
Ha = μ > 11 minutes
x̄ = 11.5 minutes
n = 1000
σ = 2.3 minutes
α = 0.05
Statistical Test
Since we have a sample of 1000, which is greater than 30, and we have a sample mean, standard deviation, and want to compare the sample mean to a population value ,so we acn use Z-test.

rejection criteria is z_statistics>critical value, fail to accept Ho

Formula Z=(x̄ - μ)/(σ/√n)

Where: x̄ = sample mean (11.5 minutes), μ = population mean (11 minutes), σ = population standard deviation (2.3 minutes), n = sample size (1000).

In [4]:
import math

In [5]:
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955

### Determine Critical Value or P-Value:
Since we're conducting a one-tailed test (since we're testing if the average time is more than 11 minutes), we will look up the crirical value for a z-score corresponding to an alpha level of 0.05 from z-distribution table.

corresponds to an alpha level of 0.05.
Critical value=1.645

## Conslusion:
Since our calculated z-score (6.88) is much greater than the critical value (1.645), we reject the null hypothesis. We have sufficient evidence to conclude that the average time working mothers spend talking to their children is indeed more than 11 minutes per day .

## Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

### Null Hypothesis
#### Ho = μ ≥ 5 minutes
### Alternative Hypothesis
#### Ha = μ < 5 minutes
#### Sample mean(x̄) = 4.6 minutes
#### Sample size(n) = 40
#### Sample Standard deviation(σ) = 0.8 minutes
#### Population mean(μ) = 5 minutes
#### Significance level(α) = 0.05(5%)
#### Degree of freedom(n-1) = 39

### Statistical Test:
#### Here the population std is unknown, so we can use t-test for this.
##### If the sample size is small (typically <30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.
t_test_statistic=(x̄ - μ) / (σ / √n)

In [6]:
t_test_statistic=(4.6 - 5) / (0.8 / np.sqrt(40))
print('t_statistic:',t_test_statistic)

t_statistic: -3.162277660168382


#### The function t.ppf(alpha, df) calculates a critical value from the t-distribution based on the provided significance level (alpha) and degrees of freedom (df).
#### The term "percent point function" (ppf) means it gives you a value from the distribution such that if you randomly select a value from the distribution, there's a certain probability it will be less than or equal to that value.
#### In simpler terms, it helps to find a specific value in the t-distribution where a certain percentage of the distribution falls below that value.

In [7]:
from scipy.stats import t
critical_value = t.ppf(0.05, 39)

print("Critical value:", critical_value)

Critical value: -1.6848751194974


Since our calculated t-value (-3.16) is less than the critical t-value (-1.685), we fail to accept the null hypothesis.
There is enough evidence to support the claim that the average wait time for customers is less than 5 minutes at a significance level of 0.05.